site/app/controllers/grading/ElectronicGraderController.php

```diff
@@ -1384,6 +1388,45 @@ private function getStats($gradeable, &$sections, $graders=array(), $total_users
             }
         }
     }
+
+    /**
+     * Check if the current user is allowed to grade the given gradeable/user pair
+     * @param string $gradeable_id
+     * @param string $user_id
+     * @return bool
+     */
+    public function canIGradeThis(string $gradeable_id, string $user_id): bool {
```

Comment:
Have this just accept a gradeable object. All instances of usage of this function, either the query it runs has been called before it or is one the next (or second next) line.